src/hybrid_failure_discovery/envs/conveyorbelt_env.py

```diff
@@ -0,0 +1,247 @@
+        self.render_mode = render_mode
+        self.observation_dim = 6
+        self._last_inserted_value: float | None = None
+        self.action_dim = 3  # Now only actions 0, 1, 2 are valid
+        super().__init__(seed)
+
+        self._step_count = 0
+
+    def _create_action_space(self):
+        return range(self.action_dim)
+
+    def _get_obs(self) -> ConveyorBeltState:
+        assert self._current_state is not None
+        return self._current_state
+
+    def get_initial_states(self) -> EnumSpace[ConveyorBeltState]:
+        assert self._np_random is not None
+        init_values = self._np_random.random(self.observation_dim).astype(np.float32)
+        init_state = ConveyorBeltState(values=init_values)
+        return EnumSpace([init_state])
+
+    def get_next_states(
+        self, state: ConveyorBeltState, action: ConveyorBeltAction
+    ) -> EnumSpace[ConveyorBeltState]:
+        # print(f"[DEBUG] Action taken: {action.index}")
+
+        next_values = state.values.copy()
+        assert self._np_random is not None
+
+        if action.index == 0:
+            # print("[DEBUG] No-op: Do nothing")
+            new_value = None
+
+        elif action.index == 1:
+            new_value = float(self._np_random.random())
+            # print("[DEBUG] Shift left and load new box on the right")
+            next_values[:-1] = next_values[1:]
+            next_values[-1] = new_value
+            # print(f"[DEBUG] Loaded new value: {new_value:.2f}")
+
+        elif action.index == 2:
+            new_value = float(self._np_random.random())
+            # print("[DEBUG] Shift right and load new box on the left")
+            next_values[1:] = next_values[:-1]
+            next_values[0] = new_value
+            # print(f"[DEBUG] Loaded new value: {new_value:.2f}")
+
+        else:
+            print(f"[WARNING] Invalid action index: {action.index}")
+
+        self._last_inserted_value = new_value
+        next_state = ConveyorBeltState(values=next_values.astype(np.float32))
+        # print(f"[DEBUG] State after:  {next_state.values}")
+        return EnumSpace([next_state])
+
+    def actions_are_equal(
+        self, action1: ConveyorBeltAction, action2: ConveyorBeltAction
+    ) -> bool:
+        return action1.index == action2.index
+
+    def _get_reward_and_termination(
+        self,
+        state: ConveyorBeltState,
+        action: ConveyorBeltAction,
+        next_state: ConveyorBeltState,
+    ) -> tuple[float, bool]:
+        return 0.0, False
+
+    def _render_state(
+        self, state: ConveyorBeltState
+    ) -> RenderFrame | list[RenderFrame] | None:
+        if self.render_mode != "rgb_array":
+            raise NotImplementedError(f"Unsupported render mode: {self.render_mode}")
+
+        fig, ax = plt.subplots(figsize=(10, 3))
+        ax.set_xlim(0, 5)
+        ax.set_ylim(0, 2)
+        ax.axis("off")
+        ax.set_title("Conveyor Belt State: 6 Boxes")
+
+        box_count = 6
```

Comment:
for all these "magic numbers", can we create and use an environment scene spec instead? check out the examples in hovercraft and blocks for reference